### PR TITLE
Make history source enum

### PR DESCRIPTION
**Pull request by eivindjahren**
No longer need this enum to be a CEnum

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/sched/history_source.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from enum import Enum
+
+
+class HistorySource(Enum):
+    REFCASE_SIMULATED = 1
+    REFCASE_HISTORY = 2
+    HISTORY_SOURCE_INVALID = 10
```
Comment (yngve-sk) [inline src/ert/_c_wrappers/sched/history_source.py:7]: do we still need the int values?

Comment (eivindjahren) [reply src/ert/_c_wrappers/sched/history_source.py:7]: I don't think so, just didn't want to change it, but I can :). Also, I think its time to remove the INVALID option

Comment (eivindjahren) [reply src/ert/_c_wrappers/sched/history_source.py]: Changed now

**Review by yngve-sk — APPROVED** [review]
🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

**Pull request merged**